packages/builder/src/registry.ts

```diff
@@ -37,6 +37,12 @@ export abstract class CannonRegistry {
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   async getMetaUrl(serviceRef: string, chainId: number): Promise<string | null> {
+    // Check if its an ipfs hash / url, if so we make sure to remove any incorrectly appended presets (like @main);
+    if (serviceRef.startsWith('@')) {
+      const result = serviceRef.replace(':', '://').replace('@', '');
```

Comment:
code here is exactly the same as the `getUrl` function.

I do not think this change should be implemented, reasoning below.

if it is determined that the meta url should match the plain url for a (ex) `@ipfs:Qmarst`, then its a change of functonality and we probably need to do some more testing

however, I would not reccomend this change because previously the funtionality was just to return null because, if you think about it, if all you have is just an IPFS hash, then there isnt a way to resolve the metadata corresponding to this ipfs hash. to attempt to do so could result in bugs/crashing when apps expect metadata to come back in one format, and it turns out to be another.